### PR TITLE
Update array-manager.md

### PR DESCRIPTION
**Pull request by ChristopheFerreboeuf**
## Purpose of this pull request

This pull request adds information about use of ArrayManager. It intends to improve the understanding of the class by adding some more examples.

Main changes : 

- Adds information and examples for use of findPaths with optional arguments.
- Adds information about array modification and example using merge and set.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/array-manager.html
- https://devdocs.magento.com/guides/v2.3/extension-dev-guide/framework/array-manager.html

## Links to Magento source code

- {{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Stdlib/ArrayManager.php
- {{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Related.php

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 group: php-developer-guide
 title: Array Manager
+contributor_name: Christophe Ferreboeuf
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:4]: Please remove the contributor items as this is not a new topic

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 2]
```diff
@@ -14,7 +16,8 @@ The library is primarily used to handle data from UI components within [DataProv
 |--- |--- |
 | `exists` | Checks if the node exists in a given associative array |
 | `find` | Finds a node in a nested array and saves its index and parent node reference |
-| `findPaths` | Gets matching paths for elements with specified indexes. |
+| `findPath` | Get first matching path for elements with specified indexes. It calls PHP `reset` function on `findPaths` results |
+| `findPaths` | Gets matching paths for elements with specified indexes. `$startPath` permits to perform a search starting from specific path. `$internalPath` permits to retrieve only elements that are situated under the specified path. |
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:20]:
```suggestion
| `findPaths` | Gets matching paths for elements with specified indexes. `$startPath` is used to define a search starting from specific path. `$internalPath` retrieves only elements that are situated under the specified path. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 3]
```diff
@@ -24,123 +27,171 @@ The library is primarily used to handle data from UI components within [DataProv
 | `set` | Sets the value into a node and returns modified data. |
 | `slicePath` | Retrieves a slice of the specified path. |
 
-### Example 1
+Suppose you have the following nested array:
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:30]:
```suggestion
Take the following nested array as an example :
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 4]
```diff
@@ -24,123 +27,171 @@ The library is primarily used to handle data from UI components within [DataProv
...
+[
+  0 => "response/items"
+  1 => "error/items"
+  2 => "response/result/items"
+]
+```
 
-        ...
-    }
+If we use `findPaths` with specifying `$startPath`, the function will return paths that contain `items` under the path `response`.
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:92]:
```suggestion
If we use `findPaths` byspecifying `$startPath`, the function will return paths that contain `items` under the path `response`.
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 5]
```diff
@@ -24,123 +27,171 @@ The library is primarily used to handle data from UI components within [DataProv
...
-        ]
-    ]
+[
+  0 => "response/items"
+  1 => "response/result/items"
 ]
 ```
 
-You can use the  `Magento\Framework\Stdlib\ArrayManager` library to access items in the array:
+If we use `findPaths` with specifying `$internalPath`, the function will return paths that contain `items` that are children of element `result`.
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:109]:
```suggestion
If we use `findPaths` by specifying `$internalPath`, the function will return paths that contain `items` that are children of element `result`.
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 6]
```diff
@@ -24,123 +27,171 @@ The library is primarily used to handle data from UI components within [DataProv
...
 ```php
 ...
 
-$this->arrayManager->populate('response/result/items', $data)
+$this->arrayManager->populate('response/result/headers', $data)
+
+...
+```
+
+It is possible to modify array in different manners for several values, `merge` is a good option but just one value,
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:137]:
```suggestion
It is possible to modify an array in different ways for several values, `merge` is one option but for just one value,
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great update to the topic

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md (hunk 1 above)
Comment (ChristopheFerreboeuf) [reply src/guides/v2.3/extension-dev-guide/framework/array-manager.md:4]: I spent 4h checking different scenarios as the documentation in the code itself is really sparse, I refactored most of the article as the examples were mostly copy-paste from existing code and I need to remove the bit of recognition this work can bring me. Sorry but it does not interest me. You can do whatever you want with this PR but I will not remove my contribution items.

**Comment by dobooth** [comment]
Closed as abandoned.

**Pull request closed without merging**